src/MOD_MESH.F90

```diff
@@ -18,7 +18,7 @@ MODULE MOD_MESH
 
 TYPE T_MESH
 integer                                     :: nod2D           ! the number of 2D nodes
-real(kind=WP)                               :: ocean_area
+real(kind=WP)                               :: ocean_area, ocean_areawithcav
```

Comment:
in case of no cavity ocean_area ==ocean_areawithcav

but in case of cavity: we do restoring only over the surface ocean part, where there is no cavity (ocean_area) but the balancing of the freshwater we do over the entire surface of the ocean including also the cavity surface